src/aws-lambda/segment-personalize-destination/segment-personalize-destination.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import json
+import boto3
+import os
+import requests  # Needed for Segment Events REST APIs
+import dateutil.parser as dp
+import logging
+
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+connections_endpoint_url = "https://api.segment.io/v1"
+connections_source_api_key = os.environ['connections_source_write_key']
+
+def api_post(url, key, payload):
+    myResponse = requests.post(url,auth=(key, ''), json=payload)
+    if(myResponse.ok):
+        jData = json.loads(myResponse.content)
+        return jData
+    else:
+        myResponse.raise_for_status()
+
+def set_user_traits(user_id, traits):
+    # Sends an identify call to Personas to update a user's traits
+    formatted_url = "{:s}/identify".format(connections_endpoint_url)
+    message = { "traits": traits, "userId": user_id, "type": "identify" }
+    try:
+        response = api_post(formatted_url, connections_source_api_key, message)
+    except HTTPError as error:
+        status = error.response.status_code
+        if status >= 400 and status < 500:
+            logger.error('Segment: 400 error, more than likely you sent an invalid request.')
+        elif status >= 500:
+            logger.error('Segment: There was a server error on the Segment side.')
+
+def lambda_handler(event, context):
+    if not 'personalize_tracking_id' in os.environ:
+        raise Exception('personalize_tracking_id not configured as environment variable')
+    if not 'personalize_campaign_arn' in os.environ:
+        raise Exception('personalize_campaign_arn not configured as environment variable')
+
+    logger.info("Segment Event: " + json.dumps(event))  # Remove this if using in production.
+
+    # Allow Personalize region to be overriden via environment variable. Optional.
+    runtime_params = { 'service_name': 'personalize-runtime' }
+    if 'region_name' in os.environ:
+        runtime_params['region_name'] = os.environ['region_name']
+
+    personalize_runtime = boto3.client(**runtime_params)
+    personalize_events = boto3.client(service_name='personalize-events')
```

Comment:
`personalize-events` should be tied to the same region as `personalize-runtime` client.